src/client/PlayerInfoModal.ts

```diff
@@ -0,0 +1,486 @@
+          if (parsed.success) {
+            if (vis === "public") {
+              pubStatsAgg = pubStatsAgg
+                ? this.mergePlayerStats(pubStatsAgg, parsed.data)
+                : parsed.data;
+            } else {
+              prvStatsAgg = prvStatsAgg
+                ? this.mergePlayerStats(prvStatsAgg, parsed.data)
+                : parsed.data;
+            }
+          }
+
+          const wins = Number((diffNode as any).wins ?? 0);
+          const losses = Number((diffNode as any).losses ?? 0);
+          const total = Number((diffNode as any).total ?? 0);
+          if (vis === "public") {
+            publicTotals.wins += wins;
+            publicTotals.losses += losses;
+            publicTotals.total += total;
+          } else {
+            privateTotals.wins += wins;
+            privateTotals.losses += losses;
+            privateTotals.total += total;
+          }
+        }
+      }
+    }
+
+    this.statsPublic = pubStatsAgg;
+    this.statsPrivate = prvStatsAgg;
+    this.statsAll =
+      this.statsPublic && this.statsPrivate
+        ? this.mergePlayerStats(this.statsPublic, this.statsPrivate)
+        : (this.statsPublic ?? this.statsPrivate);
+
+    this._publicTotalsCache = { ...publicTotals };
+    this._privateTotalsCache = { ...privateTotals };
+
+    this.requestUpdate();
+  }
+
+  private mergePlayerStats(a: PlayerStats, b: PlayerStats): PlayerStats {
+    const safeA = a ?? {};
+    const safeB = b ?? {};
+    const mergeArrays = (arr1?: any[], arr2?: any[]) => {
+      if (!arr1 && !arr2) return undefined;
+      if (!arr1) return arr2;
+      if (!arr2) return arr1;
+      return arr1.map((v, i) => Number(v ?? 0) + Number(arr2[i] ?? 0));
+    };
+    return {
+      attacks: mergeArrays(safeA.attacks, safeB.attacks),
+      betrayals: (safeA.betrayals ?? 0n) + (safeB.betrayals ?? 0n),
+      boats: { ...(safeA.boats ?? {}), ...(safeB.boats ?? {}) },
+      bombs: { ...(safeA.bombs ?? {}), ...(safeB.bombs ?? {}) },
+      gold: mergeArrays(safeA.gold, safeB.gold),
+      units: { ...(safeA.units ?? {}), ...(safeB.units ?? {}) },
+    };
+  }
+
+  render() {
+    const flag = this.getStoredFlag();
+    const playerName = this.getStoredName();
+
+    const u = this.userMeResponse?.user;
+    const discordName = u?.username ?? "";
+    const avatarUrl = u?.avatar
+      ? `https://cdn.discordapp.com/avatars/${u.id}/${u.avatar}.${u.avatar.startsWith("a_") ? "gif" : "png"}`
+      : u?.discriminator !== undefined
+        ? `https://cdn.discordapp.com/embed/avatars/${Number(u.discriminator) % 5}.png`
+        : "";
+
+    const totalsByVisibility = this.computeTotalsByVisibility();
+    const visTotals = totalsByVisibility[this.visibility];
+    const wins = visTotals?.wins ?? 0;
+    const losses = visTotals?.losses ?? 0;
+    const gamesPlayed = visTotals?.total ?? 0;
+    const winRate =
+      gamesPlayed === 0 ? "0.0" : ((wins / gamesPlayed) * 100).toFixed(1) + "%";
```

Comment:
This is how the leaderboard calculates `win:loss` ratio:

```suggestion
    const wlr = wins === 0 ? 0 : losses === 0 ? wins : wins / losses;
```